presto-pinot/src/main/java/com/facebook/presto/pinot/PinotClusterInfoFetcher.java

```diff
@@ -0,0 +1,199 @@
+
+import java.io.IOException;
+import java.net.SocketException;
+import java.net.UnknownHostException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+
+import static com.facebook.presto.pinot.PinotErrorCode.PINOT_FAILURE_GETTING_TABLE;
+import static org.apache.pinot.common.config.TableNameBuilder.extractRawTableName;
+
+/**
+ * This class handles fetching necessary information from the Pinot cluster, including table schema, broker host, routing table, etc
+ */
+public class PinotClusterInfoFetcher
+{
+    private static final String APPLICATION_JSON = "application/json";
+
+    private static final String GET_ALL_TABLES_API_TEMPLATE = "http://%s/tables";
+    private static final String TABLE_SCHEMA_API_TEMPLATE = "http://%s/tables/%s/schema";
+    private static final String ROUTING_TABLE_API_TEMPLATE = "http://%s/debug/routingTable/%s";
+    private static final String TIME_BOUNDARY_API_TEMPLATE = "http://%s/debug/timeBoundary/%s";
+
+    private static final Logger log = Logger.get(PinotClusterInfoFetcher.class);
+    private static final CloseableHttpClient httpClient = HttpClients.createDefault();
+    private final String controllerUrl;
+    private final String zkServers;
+    private DynamicBrokerSelector dynamicBrokerSelector;
+    private String instanceId = "Presto_pinot_master";
+    private ObjectMapper objectMapper = new ObjectMapper();
+
+    @Inject
+    public PinotClusterInfoFetcher(PinotConfig pinotConfig)
+            throws SocketException, UnknownHostException
+    {
+        this(pinotConfig.getZkUrl(), pinotConfig.getPinotCluster(), pinotConfig.getControllerUrl());
+    }
+
+    public PinotClusterInfoFetcher(String zkUrl, String pinotCluster, String controllerUrl)
+            throws SocketException, UnknownHostException
+    {
+        log.info("Trying to init PinotClusterInfoFetcher with Zookeeper: %s, PinotCluster %s, ControllerUrl: %s.", zkUrl, pinotCluster, controllerUrl);
+        zkServers = zkUrl + "/" + pinotCluster;
+        instanceId = instanceId + "_" + NetUtil.getHostAddress();
+        this.controllerUrl = controllerUrl;
+    }
+
+    public static String sendHttpGet(final String url)
+            throws Exception
+    {
+        HttpUriRequest request = RequestBuilder.get(url).setHeader(HttpHeaders.CONTENT_TYPE, APPLICATION_JSON)
+                .build();
+        return httpClient.execute(request, getStringResponseHandler());
+    }
+
+    private static ResponseHandler<String> getStringResponseHandler()
+    {
+        return new ResponseHandler<String>()
+        {
+            @Override
+            public String handleResponse(HttpResponse response)
+                    throws ClientProtocolException, IOException
+            {
+                int status = response.getStatusLine().getStatusCode();
+                HttpEntity entity = response.getEntity();
+                if (status >= 200 && status < 300 && entity != null) {
```

Comment:
You might use something like `HttpStatus.OK.code()`. Check `HttpPageBufferClient` as an example.